pkg/upgrade/v1_18_0.go

```diff
@@ -0,0 +1,75 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	log "github.com/sirupsen/logrus"
+)
+
+func upgrade1_18_0(ctx context.Context, client client.Client, jaeger v1.Jaeger) (v1.Jaeger, error) {
+	collectorDeprecatedFlags := []deprecationFlagMap{
+		{
+			from: "collector.port",
+			to:   "collector.tchan-server.host-port",
+		},
+		{
+			from: "collector.http-port",
+			to:   "collector.http-server.host-port",
+		},
+		{
+			from: "collector.grpc-port",
+			to:   "collector.grpc-server.host-port",
+		},
+		{
+			from: "collector.zipkin.http-port",
+			to:   "collector.zipkin.host-port",
+		},
+		{
+			from: "admin-http-port",
+			to:   "admin.http.host-port",
+		},
+	}
+
+	agentFlags := []string{
+		"collector.host-port",
+		"reporter.tchannel.discovery.conn-check-timeout",
+		"reporter.tchannel.discovery.min-peers",
+		"reporter.tchannel.host-port",
```

Comment:
Does the operator configure `--reporter.grpc.host-port` when the channel flags are removed?

Reply:
Oh no, not at this time.